packages/flutter/lib/src/material/text_field.dart

```diff
@@ -508,7 +508,9 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
     // assert(debugCheckHasMaterialLocalizations(context));
     assert(debugCheckHasDirectionality(context));
     final ThemeData themeData = Theme.of(context);
-    final TextStyle style = widget.style ?? themeData.textTheme.subhead;
+    final TextStyle style = widget.style != null
+      ? themeData.textTheme.subhead.merge(widget.style)
+      : themeData.textTheme.subhead;
```

Comment:
@HansMuller This seems to fix the issue and does not cause any tests to fail.  Do you think it's safe then?

Reply:
I think it should be safe, yes. TextStyle.merge() can handle a null parameter so this can be a one-liner.